skip_unique_checks storage param (#23214)
cockroachdb/docs: src/current/v26.2/alter-table.md

`ALTER TABLE ... RESET {storage parameter}` reverts the value of a storage parameter on a table to its default value.

{{site.data.alerts.callout_info}}

> To reset a storage parameter on an existing index, use [`ALTER INDEX ... RESET (storage_parameter)`]({% link {{ page.version.version }}/alter-index.md %}#reset) if the parameter can be altered. Otherwise, you must drop and [recreate the index without the storage parameter]({% link {{ page.version.version }}/with-storage-parameter.md %}).

Reviewer: "can be altered" should probably link to the table. Not sure what the best target for that link would be.

`ALTER TABLE ... SET {storage parameter}` sets a storage parameter on an existing table.

{{site.data.alerts.callout_info}}

> To set a storage parameter on an existing index, use [`ALTER INDEX ... SET (storage_parameter)`]({% link {{ page.version.version }}/alter-index.md %}#set) if the parameter can be altered. Otherwise, you must drop and [recreate the index with the storage parameter]({% link {{ page.version.version }}/with-storage-parameter.md %}).

Reviewer: "can be altered" should probably link to the table. Not sure what the best target for that link would be.

To ensure that the uniqueness constraint is enforced properly across regions when rows are inserted, or the `email` column of an existing row is updated, the database needs to do the following additional work when indexes are partitioned:

1. Run a one-time-only validation query to ensure that the existing data in the table satisfies the unique constraint.

> 1. Thereafter, the [optimizer]({% link {{ page.version.version }}/cost-based-optimizer.md %}) will automatically add a "uniqueness check" when necessary to any [`INSERT`]({% link {{ page.version.version }}/insert.md %}), [`UPDATE`]({% link {{ page.version.version }}/update.md %}), or [`UPSERT`]({% link {{ page.version.version }}/upsert.md %}) statement affecting the columns in the unique constraint. 

Reviewer: ```suggestion
1. By default, the [optimizer]({% link {{ page.version.version }}/cost-based-optimizer.md %}) performs a "uniqueness check" when handling an [`INSERT`]({% link {{ page.version.version }}/insert.md %}), [`UPDATE`]({% link {{ page.version.version }}/update.md %}), or [`UPSERT`]({% link {{ page.version.version }}/upsert.md %}) statement affecting the columns in the unique constraint. 
```

~~~

{{site.data.alerts.callout_info}}

> The optimizer does not perform uniqueness checks when you use `DEFAULT gen_random_uuid()` on columns in `REGIONAL BY ROW` tables. CockroachDB assumes uniqueness based on the way this column generates [`UUIDs`]({% link {{ page.version.version }}/uuid.md %}#create-a-table-with-auto-generated-unique-row-ids). To enable this check, you can modify the `sql.optimizer.uniqueness_checks_for_gen_random_uuid.enabled` [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}). Note that while there is virtually no chance of a [collision](https://wikipedia.org/wiki/Universally_unique_identifier#Collisions) occurring when enabling this setting, it is not truly zero.

Reviewer: ```suggestion
By default, the optimizer does not perform uniqueness checks when you use `DEFAULT gen_random_uuid()` on columns in `REGIONAL BY ROW` tables. CockroachDB assumes uniqueness based on the way this column generates [`UUIDs`]({% link {{ page.version.version }}/uuid.md %}#create-a-table-with-auto-generated-unique-row-ids). To enable this check, you can modify the `sql.optimizer.uniqueness_checks_for_gen_random_uuid.enabled` [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}). Note that while there is virtually no chance of a [collision](https://wikipedia.org/wiki/Universally_unique_identifier#Collisions) occurring when enabling this setting, it is not truly zero.
```

- If `crdb_region` is included in the index definition, a [`UNIQUE` index]({% link {{ page.version.version }}/unique.md %}) will enforce uniqueness on the set of columns, just like it would in a non-partitioned table.
- If `crdb_region` is excluded from the index definition, that serves as a signal that CockroachDB should enforce uniqueness on only the columns in the index definition.

> In the latter case, the index alone cannot enforce uniqueness on columns that are not a prefix of the index columns, so any time rows are [inserted]({% link {{ page.version.version }}/insert.md %}) or [updated]({% link {{ page.version.version }}/update.md %}) in a `REGIONAL BY ROW` table that has an implicitly partitioned `UNIQUE` index, the [optimizer]({% link {{ page.version.version }}/cost-based-optimizer.md %}) will add uniqueness checks by default. These checks can be skipped for unique, implicitly partitioned indexes by setting the `skip_unique_checks` [storage parameter]({% link {{ page.version.version }}/with-storage-parameter.md %}#index-parameters) to `true`. 

Reviewer: ```suggestion
In the latter case, the index alone cannot enforce uniqueness on columns that are not a prefix of the index columns, so any time rows are [inserted]({% link {{ page.version.version }}/insert.md %}) or [updated]({% link {{ page.version.version }}/update.md %}) in a `REGIONAL BY ROW` table that has an implicitly partitioned `UNIQUE` index, by default the [optimizer]({% link {{ page.version.version }}/cost-based-optimizer.md %}) performs uniqueness checks. These checks can be skipped for unique, implicitly partitioned indexes by setting the `skip_unique_checks` [storage parameter]({% link {{ page.version.version }}/with-storage-parameter.md %}#index-parameters) to `true`. 
```
